compiler/compile/OMRCompilation.cpp

```diff
@@ -294,9 +294,13 @@ OMR::Compilation::Compilation(
    _gpuPtxCount(0),
    _bitVectorPool(self()),
    _typeLayoutMap((LayoutComparator()), LayoutAllocator(self()->region())),
-   _target(TR::Compiler->target),
    _tlsManager(*self())
    {
+   if (self()->compileRelocatableCode())
```

Comment:
I know that this is already a problem here, but I do want to point out that calling `self()` in a constructor relies on undefined behaviour. Is there any chance this could be avoided?

Reply:
Thanks for pointing this out. I kept forgetting about this and it doesn't help when the rest of the code there still uses `self()` LOL.
I'm surprised the other self() instances didn't break anything..